redhat-access/app/controllers/redhat_access/api/machine_telemetry_api_controller.rb

```diff
@@ -51,9 +51,14 @@ def proxy_upload
         else
           resource = "uploads/#{params[:id]}"
         end
+        if request.format.json?
+          original_payload = original_payload.to_json
+        end
         if params[:file]
```

Comment:
I'll be extra pedantic here and point out that one of these tests should really be an "else if", probably: 
```
if params[:file]
...
else if request.format.json?
...
```
but it's clear enough as-is so I'm merging anyway, lol